CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 3.16.3)
+project(ThreadPool)
+
+set(CMAKE_CXX_STANDART 17)
```

Comment:
add_compiler options for a better warning analysis
https://cmake.org/cmake/help/latest/command/add_compile_options.html